src/components/fleet-management/edit-device-drawer.tsx

```diff
@@ -0,0 +1,328 @@
+import { Device, DeviceApprovalStatus, DeviceModel } from "../../generated/client";
+import strings from "../../localization/strings";
+import theme from "../../styles/theme";
+import GenericUtils from "../../utils/generic-utils";
+import PanelSubtitle from "../layout/v2/panel-subtitle";
+import PropertyBox from "../layout/v2/property-box";
+import { Close as CloseIcon } from "@mui/icons-material";
+import {
+  Button,
+  Divider,
+  Drawer,
+  FormControlLabel,
+  IconButton,
+  MenuItem,
+  Stack,
+  Switch,
+  TextField,
+  Typography
+} from "@mui/material";
+import { DatePicker, DateTimePicker } from "@mui/x-date-pickers";
+import moment from "moment";
+import { ChangeEvent, ReactNode, useState } from "react";
+
+/**
+ * Components properties
+ */
+interface Props {
+  selectedDevice?: Device;
+  deviceModels: DeviceModel[];
+  onSave: (device: Device) => Promise<void>;
+  onClose: () => void;
+}
+
+/**
+ * Edit Device Drawer component
```

Comment:
MIssing @params docs

Reply:
We haven't documented the props passed for components.